bridge-token-factory/tests/token_transfer.rs

```diff
@@ -63,161 +84,218 @@ impl From<Proof> for AugmentedProof {
-            .unwrap_json();
-    assert_eq!(alice_balance, "0");
-
-    let total_supply: String = call_json!(user, token_account_id.ft_total_supply({})).unwrap_json();
-    assert_eq!(total_supply, "0");
+    .unwrap()
+    .to_string();
+    assert_eq!(
+        token_account_id,
+        format!("\"{}.{}\"", DAI_ADDRESS, factory.id())
+    );
+    token_account_id = token_account_id[1..token_account_id.len() - 1]
+        .parse()
+        .unwrap();
+
+    let alice_balance: String = std::str::from_utf8(
+        &rt.block_on(
+            worker.view(
+                &token_account_id.parse().unwrap(),
+                "ft_balance_of",
+                json!({"account_id": user.id().to_string()})
+                    .to_string()
+                    .into_bytes(),
+            ),
+        )
+        .unwrap()
+        .result,
+    )
+    .unwrap()
+    .to_string();
+    assert_eq!(alice_balance, "\"0\"");
+
+    let total_supply: String = std::str::from_utf8(
+        &rt.block_on(worker.view(
+            &token_account_id.parse().unwrap(),
+            "ft_total_supply",
+            json!({}).to_string().into_bytes(),
+        ))
+        .unwrap()
+        .result,
+    )
+    .unwrap()
+    .to_string();
+    assert_eq!(total_supply, "\"0\"");
 
     let mut proof = Proof::default();
     proof.log_entry_data = EthLockedEvent {
         locker_address: validate_eth_address(LOCKER_ADDRESS.to_string()),
         token: DAI_ADDRESS.to_string(),
         sender: SENDER_ADDRESS.to_string(),
         amount: 1_000,
-        recipient: ALICE.parse().unwrap(),
+        recipient: user.id().to_string().parse().unwrap(),
     }
     .to_log_entry_data();
 
-    call_borsh!(user, factory_id.deposit(proof)).assert_success();
-
-    let alice_balance: String =
-        call_json!(user, token_account_id.ft_balance_of({"account_id": ALICE.to_string()}))
-            .unwrap_json();
-    assert_eq!(alice_balance, "1000");
-
-    let total_supply: String = call_json!(user, token_account_id.ft_total_supply({})).unwrap_json();
-    assert_eq!(total_supply, "1000");
-
-    call_json!(alice, token_account_id.withdraw({
-                "amount" : "100",
-                "recipient" : SENDER_ADDRESS.to_string()}), deposit=1)
-    .assert_success();
-
-    let alice_balance: String =
-        call_json!(user, token_account_id.ft_balance_of({"account_id": ALICE.to_string()}))
-            .unwrap_json();
-    assert_eq!(alice_balance, "900");
-
-    let total_supply: String = call_json!(user, token_account_id.ft_total_supply({})).unwrap_json();
-    assert_eq!(total_supply, "900");
+    assert!(&rt
+        .block_on(
+            user.call(&worker, factory.id(), "deposit")
+                .deposit(parse_near!("50 N"))
+                .max_gas()
+                .args(proof.try_to_vec().unwrap())
+                .transact()
+        )
+        .unwrap()
+        .is_success());
```

Comment:
Where this deposit value is coming from?

Reply:
If don't set any deposit appear following error: ExecutionError("Smart contract panicked: panicked at 'assertion failed: env::attached_deposit() >=\\n    required_deposit + self.bridge_token_storage_deposit_required', src/lib.rs:329:9")

Changed 50N to 1N (https://github.com/aurora-is-near/rainbow-token-connector/pull/159/commits/893272e94e78ab51e29d02e870eb2c44af54eae6). We should deposit enough near tokens for our storage. But I don't see any functions in contract which say how much is enough and calculation is not very simple. So I decide that for this test is ok to transfer just some amount of tokens and don't make accurate calculation.

Reply:
fixed: https://github.com/aurora-is-near/rainbow-token-connector/pull/159/commits/87ee9af208bad1f1faafcf4ba8e9aaf48e782e82

If deposit wasn't set the next const were used: https://docs.rs/near-sdk-sim/latest/near_sdk_sim/constant.STORAGE_AMOUNT.html

Reply:
@olga24912 I feel 50 NEAR is too huge a value for that deposit, isn't it? I believe it should work with a much smaller value.